src/StudentsController.ts

```diff
@@ -165,10 +169,53 @@ export class StudentsController {
         await this.#studentsCollection.update(oldDoc, student.toJSON());
     }
 
+    public async pseudonymizeStudent(student: Student): Promise<Student> {
+        // Only keep the id of the student
+        const oldDoc = await this.#studentsCollection.read(student.id.toString());
+        const pseudonymizedDoc = {
+            id: oldDoc.id
+        };
+
+        await this.#studentsCollection.update(oldDoc, pseudonymizedDoc);
+
+        return Student.from(pseudonymizedDoc);
+    }
+
     public async deleteStudent(student: Student): Promise<void> {
+        if (student.correspondingRelationshipId) {
+            let relationship = await this.getRelationship(student.correspondingRelationshipId);
+
+            if (relationship.status === RelationshipStatus.Pending) {
+                await this.services.transportServices.relationships.rejectRelationship({ relationshipId: student.correspondingRelationshipId.toString() });
+            } else if (relationship.status === RelationshipStatus.Active) {
+                await this.services.transportServices.relationships.terminateRelationship({ relationshipId: student.correspondingRelationshipId.toString() });
+            }
+
+            relationship = await this.getRelationship(student.correspondingRelationshipId);
+
+            if (
+                relationship.status === RelationshipStatus.Terminated ||
+                relationship.status === RelationshipStatus.Rejected ||
+                relationship.status === RelationshipStatus.Revoked ||
+                relationship.status === RelationshipStatus.DeletionProposed
+            ) {
```

Comment:
I'm not a fan of this 

- doing something
- reloading
- doing something

IMO we should switch ONCE over the relationship status and then do

- reject AND decompose when pending
- terminate AND decompose when active
- only decompose with the rest

the switch also makes sure that you decompose the relationship no mattern in which status.